### PR TITLE
Add pagination to update available

### PR DESCRIPTION
**Pull request by acosferreira**
# Description

Add pagination to the list of update available for device

Fixes # (THEEDGE-2821)

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/services/mock_services/devices.go [hunk 1]
```diff
@@ -5,283 +5,282 @@
...
+	ret := m.ctrl.Call(m, "GetDevicesView", limit, offset, tx)
+	ret0, _ := ret[0].(*models.DeviceViewList)
 	ret1, _ := ret[1].(error)
 	return ret0, ret1
 }
 
-// GetDeviceDetails indicates an expected call of GetDeviceDetails.
-func (mr *MockDeviceServiceInterfaceMockRecorder) GetDeviceDetails(device interface{}) *gomock.Call {
+// GetDevicesView indicates an expected call of GetDevicesView
+func (mr *MockDeviceServiceInterfaceMockRecorder) GetDevicesView(limit, offset, tx interface{}) *gomock.Call {
```
Comment (thom-at-redhat) [inline pkg/services/mock_services/devices.go:78]: Should this be "GetDevicesView(limit, offset int, tx" to match line 69?

Comment (thom-at-redhat) [inline pkg/services/mock_services/devices.go:78]: Should all limits/offsets be the same (i.e. should this line be "GetDevicesView(limit, offset int, tx" to match line 69)?

**Review by thom-at-redhat — CHANGES_REQUESTED** [review]

**Review by omaciel — CHANGES_REQUESTED** [review]
I don't see unittests for the actual changes that would prove that pagination is actually working. As most of the updates to existing tests are mocking the response, I feel that we're not actually testing the functionality of the modified functions. A unittest for `GetDeviceImageInfoByUUID` for example that returns the expected number of items is exactly what I'm looking for. The same goes for every function that has been modified to include pagination arguments.

**Review by omaciel — DISMISSED** [review]
/lgtm

**Review by thom-at-redhat — APPROVED** [review]

**Comment by omaciel** [comment]
@mergify queue edge-api

**Comment by omaciel** [comment]
@mergifyio refresh

**Comment by omaciel** [comment]
@mergifyio queue

**Comment by omaciel** [comment]
@mergifyio refresh edge-api

**Comment by omaciel** [comment]
@mergifyio queue edge-api

**Comment by omaciel** [comment]
@mergifyio refresh

**Review by matoval — APPROVED** [review]

**Comment by matoval** [comment]
@mergifyio refresh

**Comment by ldjebran** [comment]
@Mergifyio requeue

**Comment by thom-at-redhat** [comment]
@mergifyio rebase

**Comment by thom-at-redhat** [comment]
@mergifyio requeue

**Review by matoval — APPROVED** [review]

**Review by thom-at-redhat — APPROVED** [review]

**Pull request merged**